### PR TITLE
Add functional tests for imports api

### LINKED ISSUE #852: requests to /v1/imports/images/{operation_id}/image_config do not return config digest after upload
**Is this a request for help?**:

NO
---

**Is this a BUG REPORT or a FEATURE REQUEST?** (choose one):

BUG REPORT

**Version of Anchore Engine and Anchore CLI if applicable**:
anchore_engine-0.9.0

**What happened**:
POSTed image config to `/v1/imports/images/{operation_id}/image_config` and received 200 response code. Subsequently tried to GET image config from `/v1/imports/images/{operation_id}/image_config` and received empty list response with status code 200. 

**What did you expect to happen**:
I expected that GET request to `/v1/imports/images/{operation_id}/image_config` would return digest of uploaded config.

**Any relevant log output from /var/log/anchore**:
```
[service:enterprise-apiext] 2021-01-19 20:45:58+0000 [-] "172.26.0.1" - - [19/Jan/2021:20:45:58 +0000] "POST /v1/imports/images/244f0c83660b4bfa932b394104a20282/image_config HTTP/1.1" 200 123 "-" "python-requests/2.23.0"
[service:enterprise-apiext] 2021-01-19 20:45:58+0000 [-] "172.26.0.1" - - [19/Jan/2021:20:45:58 +0000] "GET /v1/imports/images/244f0c83660b4bfa932b394104a20282/image_config HTTP/1.1" 200 3 "-" "python-requests/2.23.0"
```

**What docker images are you using**:
```
anchore/enterprise-dev:latest
anchore/anchore-engine-dev:latest
```

**How to reproduce the issue**:
Do as described above

**Anything else we need to know**:
N/A

### PR DESCRIPTION
**Pull request by Vijay-P**
**What this PR does / why we need it**:

Current test for imports api is not integrated into test suite. The included tests also have greater coverage.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>)(, fixes #<issue_number, ...)` format, will close the issue when PR is merged*: fixes #:

fixes #852 

**Special notes**:

Added `incremental` pytest mark. The tests do not share state. However, if one of the earlier stage tests fail, then it's likely that the later tests will fail. These tests cover a sequence of actions:
1. Creating an import operation
2. Deleting an import operation
3. Listing imports 
4. Uploading Dockerfile
5. Listing uploaded Dockerfiles
6. Uploading Syft SBOM
7. Listing uploaded Syft SBOMs
8. Uploading image manifest
9. Listing uploaded image manifests
10. Uploading image config
11. Listing uploaded image configs
12. Steps 1, 4, 6, 8, and 10 in addition to uploading an image and verifying that the image has been uploaded via /images API.

### CODE REVIEW

File: anchore_engine/services/apiext/api/controllers/image_imports.py [hunk 1]
```diff
@@ -228,7 +228,7 @@ def list_import_image_configs(operation_id):
         client = internal_client_for(
             CatalogClient, userId=ApiRequestContextProxy.namespace()
         )
-        resp = client.list_import_content(operation_id, "parent_manifest")
+        resp = client.list_import_content(operation_id, "image_config")
```
Comment (Vijay-P) [inline anchore_engine/services/apiext/api/controllers/image_imports.py:231]: this copypasta was causing the bug

File: tests/functional/services/api/imports/test_import.py [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+    logger.info("Deleting import operation.")
+    delete_response = http_del(["imports", "images", import_response.body["uuid"]])
+    if delete_response.code == 200:
+        logger.info("Deletion successful.")
+    else:
+        logger.info(f"Deletion failed! Response: {delete_response.body}")
+
+
+@pytest.mark.incremental
+@pytest.mark.parametrize(
```
Comment (Vijay-P) [inline tests/functional/services/api/imports/test_import.py:35]: this way we can parametrize the entire class with different syft-sbom+dockerfile combinations in the future.

File: tox.ini [hunk 3]
```diff
@@ -11,5 +11,5 @@ passenv=ANCHORE_* DOCKER_* NO_PROXY
 commands=
     coverage erase
     coverage run -m pytest --junitxml=.tox/results/tox/junit-{envname}.xml {posargs}
-    coverage report --include=*anchore_engine* --omit=*tests*
-    coverage html --include=*anchore_engine* --omit=*tests*
+    - coverage report --include=*anchore_engine* --omit=*tests*
```
Comment (Vijay-P) [inline tox.ini:14]: there are some occasions where coverage report may be empty. adding the `-` allows tox to exit without error code

**Comment by Vijay-P** [comment]
scripts/tests/image_import/test_import.py also waits for import to complete and gets vulnerabilities. Wasn't sure if it's necessary to include those steps in this test.

File: tests/functional/services/api/imports/__init__.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    :rtype: str
+    """
+    directory, _ = os.path.splitext(module_path)
+    filename = os.path.join(directory, filename)
+    with open(filename) as f:
+        text = f.read()
+    return text
+
+
+def http_post_bytes(
```
Comment (nightfurys) [inline tests/functional/services/api/imports/__init__.py:37]: My only recommendation is move this to utils so it can be utilized by other tests

**Review by nightfurys — APPROVED** [review]

File: tests/functional/services/api/imports/__init__.py (hunk 4 above)
Comment (Vijay-P) [reply tests/functional/services/api/imports/__init__.py]: done

**Review by Toure — APPROVED** [review]

**Pull request merged**